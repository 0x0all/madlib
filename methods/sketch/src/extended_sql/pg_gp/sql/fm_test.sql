\i sketches.sql
set search_path to "$user",public,madlib;

-- tests for "little" tables using sortasorts
select fmcount(R.i)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);

select fmcount(CAST('2010-10-10' As date) + CAST((R.i || ' days') As interval))
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);

select fmcount(R.i::float)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);

select fmcount(R.i::text)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);


-- tests for "big" tables
select fmcount(T.i)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);

select fmcount(CAST('2010-10-10' As date) + CAST((T.i || ' days') As interval))
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);

select fmcount(T.i::float)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);

select fmcount(T.i::text)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);

-- tests for all-NULL column
select fmcount(NULL::integer) from generate_series(1,10000) as R(i);
