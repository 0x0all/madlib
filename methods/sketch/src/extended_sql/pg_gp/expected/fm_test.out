set search_path to "$user",public,madlib;
-- tests for "little" tables using sortasorts
select fmcount(R.i)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);
 fmcount 
---------
     100
(1 row)

select fmcount(CAST('2010-10-10' As date) + CAST((R.i || ' days') As interval))
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);
 fmcount 
---------
     100
(1 row)

select fmcount(R.i::float)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);
 fmcount 
---------
     100
(1 row)

select fmcount(R.i::text)
  from generate_series(1,100) AS R(i),
       generate_series(1,3) AS T(i);
 fmcount 
---------
     100
(1 row)

-- tests for "big" tables
select fmcount(T.i)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);
 fmcount 
---------
   20897
(1 row)

select fmcount(CAST('2010-10-10' As date) + CAST((T.i || ' days') As interval))
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);
 fmcount 
---------
   20339
(1 row)

select fmcount(T.i::float)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);
 fmcount 
---------
   20897
(1 row)

select fmcount(T.i::text)
  from generate_series(1,3) AS R(i),
       generate_series(1,20000) AS T(i);
 fmcount 
---------
   20897
(1 row)

-- tests for all-NULL column
select fmcount(NULL::integer) from generate_series(1,10000) as R(i);
 fmcount 
---------
       0
(1 row)

